common/rfb/VNCSConnectionST.cxx

```diff
@@ -445,6 +450,11 @@ void VNCSConnectionST::queryConnection(const char* userName)
 
 void VNCSConnectionST::clientInit(bool shared)
 {
+  if (!server->isDesktopReady())
+    return;
+
+  desktopReady();
```

Comment:
This also looks wrong. The ordering gets confusing here.

There is a general design problem here of whose responsibility it is to call `desktopReady()`. Since it sometimes falls upon the server object to do so, the ideal design would be that it is always that which calls it.

Which might be feasible since the server gets the `clientReady()` call.